src/contracts/libraries/GPv2Transfer.sol

```diff
@@ -23,8 +23,51 @@ library GPv2Transfer {
     address internal constant BUY_ETH_ADDRESS =
         0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
 
+    /// @dev Execute the specified transfer from the specified account to a
+    /// recipient. The recipient will either receive internal Vault balances or
+    /// ERC20 token balances depending on whether the account is using internal
+    /// balances or not.
+    ///
+    /// This method is used for transferring fees to the settlement contract
+    /// when settling a single order directly with Balancer.
+    ///
+    /// @param vault The Balancer vault to use.
+    /// @param recipient The recipient for the transfer.
+    /// @param transfer The transfer to perform.
+    function transferFromAccount(
+        IVault vault,
+        address recipient,
+        Data calldata transfer
+    ) internal {
+        require(
+            address(transfer.token) != BUY_ETH_ADDRESS,
+            "GPv2: cannot transfer native ETH"
+        );
+
+        if (transfer.useInternalBalance) {
+            IVault.BalanceTransfer[] memory vaultTransfers =
+                new IVault.BalanceTransfer[](1);
+
+            IVault.BalanceTransfer memory vaultTransfer = vaultTransfers[0];
+            vaultTransfer.token = transfer.token;
+            vaultTransfer.amount = transfer.amount;
+            vaultTransfer.sender = transfer.account;
+            vaultTransfer.recipient = recipient;
+
+            vault.transferInternalBalance(vaultTransfers);
```

Comment:
Why the change in semantics here? Is is to be more gas efficient? One nice things about withdrawing the fees into external balance is that it can be used as a form of buffer.

Reply:
> Is is to be more gas efficient?

Yes.

> One nice things about withdrawing the fees into external balance is that it can be used as a form of buffer.

This is true, but the fast-path can't make use of this buffer (since it trades with user funds directly) and the full-path can use a pre-interaction to withdraw internal balances if it needs to top up its balances.

WDYT?